### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Removes deprecated code after v13 release. This purges a lot of old code which we forgot to clean up. Some explicit call outs of deprecations:

- precision
- add_eval_metrics for HF model (artifact of ICL eval)
- cross entropy deprecated fns
- validate from old metrics design 

Additionally, cleanup includes:
- remove YAHP vestiges 
- remove meta.yaml for conda, which is very out of date
- reworks callback tests to use filter from callback marks
- replaces log.warning with warnings.warn for algorithms being disabled CC: @eracah 

Additionally, non-cleanup includes:
- bug fix with new eval_metrics arg which was not set if metrics was not provided CC: @dakinggg 

Note:
- grad_accum removal will come separately because it's a bigger change

# What issue(s) does this change relate to?

[CO-1859](https://mosaicml.atlassian.net/browse/CO-1859)
[CO-1822](https://mosaicml.atlassian.net/browse/CO-1822)

[CO-1859]: https://mosaicml.atlassian.net/browse/CO-1859?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CO-1822]: https://mosaicml.atlassian.net/browse/CO-1822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/callbacks/callback_settings.py [hunk 1]
```diff
@@ -117,7 +116,10 @@
     ],
     ProgressBarLogger: [
         pytest.mark.filterwarnings(
-            r'ignore:Specifying the ProgressBarLogger via `loggers` is deprecated:DeprecationWarning')
+            r'ignore:Specifying the ProgressBarLogger via `loggers` is not recommended as.*:Warning')
```
[the remark this answers is not in the document]
Comment (mvpatel2000) [reply tests/callbacks/callback_settings.py:119]: Sometimes users do want to override certain arguments, and we decided not to flatten every arg into trainer level CC: @eracah

**Review by eracah — APPROVED** [review]
LGTM, but I mostly looked at the stuff relevant to me. Might want to get one more set of eyes on this

File: tests/models/test_hf_model.py [hunk 2]
```diff
@@ -594,17 +575,12 @@ def test_separate_eval_metrics(tiny_bert_model, tiny_bert_tokenizer):
 def test_add_eval_metrics(tiny_bert_model, tiny_bert_tokenizer):
```
Comment (dakinggg) [inline tests/models/test_hf_model.py:575]: this test is actually the same as the one above it now, and you can just drop it

**Review by dakinggg — APPROVED** [review]
LGTM, thanks for cleaning all of that up!

File: composer/metrics/nlp.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def __init__(self, vocab_size: Optional[int] = None, dist_sync_on_step: bool = F
         if vocab_size is not None:
             warnings.warn(
                 DeprecationWarning(
-                    'The vocab_size argument is deprecated and will be removed in 0.14. It is no longer needed, because the correct shape of output and target is inferred based on the number of target elements.'
+                    'The vocab_size argument is deprecated and will be removed in 0.15. It is no longer needed, because the correct shape of output and target is inferred based on the number of target elements.'
```
Comment (bandish-shah) [inline composer/metrics/nlp.py:87]: Why are we pushing this deprecation warnings out to 0.15?

Comment (mvpatel2000) [reply composer/metrics/nlp.py:87]: It is a version too early and doesn't give a 2 release period as we require.

File: composer/models/resnet/model.py [hunk 4]
```diff
@@ -92,7 +93,7 @@ def composer_resnet(model_name: str,
         weights = 'IMAGENET1K_V2'
         warnings.warn(
             DeprecationWarning(
-                'The ``pretrained`` argument for composer_resnet is deprecated and will be removed in the future. Please use ``weights`` instead.'
+                'The ``pretrained`` argument for composer_resnet is deprecated and will be removed in the future when torch 1.11 is no longer supported. Please use ``weights`` instead.'
```
Comment (bandish-shah) [inline composer/models/resnet/model.py:96]: FYI: Due to security vulnerabilities, we will likely deprecate Torch 1.11 support in the next release, minor or micro whatever comes first.

Comment (bandish-shah) [inline composer/models/resnet/model.py:96]: FYI: Due to security vulnerabilities, we will likely deprecate Torch 1.11 support in the next release, minor or micro whatever comes first.

**Review by bandish-shah — APPROVED** [review]
LGTM

**Pull request merged**